### DISCUSSION TITLE
Creating custom Properties on Listing (513)

### DISCUSSION THREAD

**Discussion by edgar-haak** [etsy/open-api#1097] [Q&A] [answered]
Hello

I'm trying to create custom Properties for a listing to create variation products.
I'm using the propertyID 513

I tried different ways (as seen in the discussions here):

URL = "/application/shops/$this->shop_id/listings/$listing_id/properties/513"
$data = array (
  'value_ids' => '',
  'values' =>
  array (
    0 => 'test133',
    1 => 'test442',
  ),

Result : There was a problem with /attributes : Attributes not valid for property
__
array (
  'value_ids' =>
  array (
  ),
  'values' =>
  array (
    0 => 'test133',
    1 => 'test442',
  ),
)

Result => Missing input parameter: [value_ids]
___
array (
  'value_ids' =>
  array (
    0 => '',
    1 => '',
  ),
  'values' =>
  array (
    0 => 'test133',
    1 => 'test442',
  ),
)

Result: Expected int value for 'value_ids' (got string)

___

what am i doing wrong ?

**Comment by seandmccarthy** [comment, marked as answer]
To create custom variations on a product we did the following:

- Create a draft listing https://developers.etsy.com/documentation/reference/#operation/createDraftListing
- Add the variations using `updateListingInventory` https://developers.etsy.com/documentation/reference/#operation/updateListingInventory

(then add images to the listing, and then use `updateListing` to set the listing `state` to active if it's ready to list).

Here's an example payload for updateListingInventory (in JSON)

```
{
    "price_on_property": [513],
    "quantity_on_property": [513],
    "sku_on_property": [513], 
    "products": [
	    {
		    "sku": "baseball-hat-pink",
		    "property_values": [
			    {
				    "property_id": 513,
				    "property_name": "Colour",
				    "value_ids": [ 1 ],
				    "values": [ "Pinkish" ]
			    }
		    ],
		    "offerings": [
			    {
				    "price": 4444.44,
				    "quantity": 10,
				    "is_enabled": true
			    }
		    ]
	    },
	    {
		    "sku": "baseball-hat-blue",
		    "property_values": [
			    {
				    "property_id": 513,
				    "property_name": "Colour",
				    "value_ids": [ 2 ],
				    "values": [ "Blue" ]
			    }
		    ],
		    "offerings": [
			    {
				    "price": 4445.44,
				    "quantity": 20,
				    "is_enabled": true
			    }
		    ]
	    }
    ]
}
```

We actually found that you don't need `value_ids` for custom values, so you can try omitting them. Supplying the `values` array was enough.

**Comment by edgar-haak** [comment]
thank you @seandmccarthy , that worked!
